### PR TITLE
[1.0.0] S3 interoperability & ULIDs

### PR DESCRIPTION
**Pull request by St4NNi**
This update introduces a multitude of major features for the 1.0.0 release, this includes:

## Changelog
Switch from UUID to ULIDs:
- ULIDs are easier to sort, contain an additional timestamp and are easier to represent / share via Websites and URLs
Add S3 compability:
- Added first iteration of S3 compability including:
- Subpaths / Paths for Objects & Collections 
- New Data proxy backend with: Data deduplication, Encryption at REST and hash validation
- Various bugfixes
- Session tokens
- Updates for Userhandling APIs
- Dependency updates
- Security fixes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,42 +1,48 @@
 [package]
 edition = "2021"
 name = "aruna_server"
-version = "1.0.0-rc.5"
+version = "1.1.0-rc.1"
```
Comment (MariusDieckmann) [inline Cargo.toml:4]: Do we really want to jump to 1.1.0?

Comment (MariusDieckmann) [inline Cargo.toml:4]:
```suggestion
version = "1.0.0"
```
we should start with 1.0.0.

Comment (St4NNi) [reply Cargo.toml]: Fixed in https://github.com/ArunaStorage/ArunaServer/pull/76/commits/bd3081fbebe63c9e408d19a46964197a4918b820

**Review by das-Abroxas — APPROVED** [review]
LGTM :heart:

**Pull request merged**